references/depth/stereo/train.py

```diff
@@ -0,0 +1,750 @@
+                image_left, image_right, disp_gt, valid_disp_mask = (x.to(device) for x in blob)
+                padder = utils.InputPadder(image_left.shape, mode=padder_mode)
+                image_left, image_right = padder.pad(image_left, image_right)
+
+                disp_predictions = module(image_left, image_right, flow_init=None, num_iters=iterations)
+                disp_pred = disp_predictions[-1][:, :1, :, :]
+                disp_pred = padder.unpad(disp_pred)
+
+                metrics, _ = utils.compute_metrics(disp_pred, disp_gt, valid_disp_mask, metrics=logger.meters.keys())
+                num_processed_samples += image_left.shape[0]
+                for name in metrics:
+                    logger.meters[name].update(metrics[name], n=1)
+    
+    num_processed_samples = utils.reduce_across_processes(num_processed_samples)
+    
+    print("Num_processed_samples: ", num_processed_samples)
+    if num_processed_samples != len(val_dataset):
+        warnings.warn(
+            f"Number of processed samples {num_processed_samples} is different"
+            f"from the dataset size {len(val_dataset)}. This may happen if"
+            "the dataset is not divisible by the batch size. Try lowering the batch size for more accurate results."
+        )
+
+    if writter is not None and args.rank == 0:
+        for meter_name, meter_value in logger.meters.items():
+            scalar_name = f"{meter_name} {header}"
+            writter.add_scalar(scalar_name, meter_value.avg, step)
+    
+    logger.synchronize_between_processes()
+    print(header, logger)
+
+
+def evaluate(model, args, writter=None, step=None):
+    val_datasets = args.test_datasets or []
+
+    if args.weights and args.test_only:
+        weights = torchvision.models.get_weight(args.weights)
+        trans = weights.transforms()
+
+        def preprocessing(image_left, image_right, disp, valid_disp_mask):
+            image_left, image_right = trans(image_left, image_right)
+            if disp is not None and not isinstance(disp, torch.Tensor):
+                disp = torch.from_numpy(disp)
+            if valid_disp_mask is not None and not isinstance(valid_disp_mask, torch.Tensor):
+                valid_disp_mask = torch.from_numpy(valid_disp_mask)
+            return image_left, image_right, disp, valid_disp_mask
+
+    else:
+        preprocessing = make_eval_transform(args)
+
+    for name in val_datasets:
+        val_dataset = make_dataset(name, args.dataset_root, transforms=preprocessing)
+        _evaluate(
+            model,
+            args,
+            val_dataset,
+            iterations=args.recurrent_updates,
+            padder_mode="kitti",
+            header=f"{name} evaluation",
+            batch_size=args.batch_size,
+            writter=writter,
+            step=step,
+        )
+
+
+def run(model, optimizer, scheduler, train_loader, logger, writer, scaler, args):
```

Comment:
NIT, instead of run, maybe we should call it `train` instead

Reply:
My only concern with calling it train was that in contains a call to the evaluation loop as well.